### PR TITLE
Revert "Fix `is_total_slice` for size-1 dimensions (#1800)"

### LINKED ISSUE #1874: Regression in v2.18.0 when setting an item in an object array
### Zarr version

v2.18.0

### Numcodecs version

0.12.1

### Python Version

3.12

### Operating System

Windows

### Installation

via `poetry install`

### Description

There seems to be a regression in v2.18.0 (compared to v2.17.2) concerning setting items in an object array.

### Steps to reproduce

Setup:

```py
import zarr
from numcodecs import MsgPack
import numpy as np

root = zarr.group()
arr = root.create_dataset(
    name="my_dataset",
    shape=0,
    dtype=object,
    object_codec=MsgPack(),
)
new_items = [
    ["A", 1],
    ["B", 2, "hello"],
]
arr_add = np.empty(len(new_items), dtype=object)
arr_add[:] = new_items
arr.append(arr_add)
arr[:] # --> array([list(['A', 1]), list(['B', 2, 'hello'])], dtype=object)
```
Problem:

```py
arr[0] = ["C", 3] # raises AttributeError: 'list' object has no attribute 'astype' (from Array._process_for_setitem)
```

Using Zarr v2.17.2, no such exception is raised, and the first item is successfully modified.

### Additional output

_No response_

### PR DESCRIPTION
**Pull request by dcherian**
This reverts commit 9d046ea.

Closes #1874 by reverting #1800

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [x] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: zarr/util.py [hunk 1]
```diff
@@ -234,17 +234,8 @@ def is_total_slice(item, shape: Tuple[int]) -> bool:
     if isinstance(item, tuple):
         return all(
             (
-                (
```
Comment (dcherian) [inline zarr/util.py:237]: Alternatively I could comment this out and add a TODO

Comment (joshmoore) [reply zarr/util.py:237]: From my side, +1 for reverting. We can also revert the revert.

**Comment by jhamman** [comment]
I lean toward failing forward on this issue. #1875 is looking good so I think we should get that out, rather than revert.

**Comment by dcherian** [comment]
As you wish :)

**Pull request closed without merging**